File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/view/ViewTitle.java
package work.socialhub.kslack.entity.view;

public class ViewTitle {
    private String type;
    private String text;
    private boolean emoji;

    public ViewTitle(String type, String text, boolean emoji) {
        this.type = type;
        this.text = text;
        this.emoji = emoji;
    }

    public ViewTitle() {
    }

    public static ViewTitleBuilder builder() {
        return new ViewTitleBuilder();
    }

    public String getType() {
        return this.type;
    }

    public String getText() {
        return this.text;
    }

    public boolean isEmoji() {
        return this.emoji;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setText(String text) {
        this.text = text;
    }

    public void setEmoji(boolean emoji) {
        this.emoji = emoji;
    }

    public static class ViewTitleBuilder {
        private String type;
        private String text;
        private boolean emoji;

        ViewTitleBuilder() {
        }

        public ViewTitle.ViewTitleBuilder type(String type) {
            this.type = type;
            return this;
        }

        public ViewTitle.ViewTitleBuilder text(String text) {
            this.text = text;
            return this;
        }

        public ViewTitle.ViewTitleBuilder emoji(boolean emoji) {
            this.emoji = emoji;
            return this;
        }

        public ViewTitle build() {
            return new ViewTitle(type, text, emoji);
        }

        public String toString() {
            return "ViewTitle.ViewTitleBuilder(type=" + this.type + ", text=" + this.text + ", emoji=" + this.emoji + ")";
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/view/ViewTitle.kt
package work.socialhub.kslack.entity.view

class ViewTitle {
    var type: String? = null
    var text: String? = null
    var isEmoji: Boolean = false

    constructor(type: String?, text: String?, emoji: Boolean) {
        this.type = type
        this.text = text
        this.isEmoji = emoji
    }

    constructor()

    class ViewTitleBuilder internal constructor() {
        private var type: String? = null
        private var text: String? = null
        private var emoji = false

        fun type(type: String?): ViewTitleBuilder {
            this.type = type
            return this
        }

        fun text(text: String?): ViewTitleBuilder {
            this.text = text
            return this
        }

        fun emoji(emoji: Boolean): ViewTitleBuilder {
            this.emoji = emoji
            return this
        }

        fun build(): ViewTitle {
            return ViewTitle(type, text, emoji)
        }

        override fun toString(): String {
            return "ViewTitle.ViewTitleBuilder(type=" + this.type + ", text=" + this.text + ", emoji=" + this.emoji + ")"
        }
    }

    companion object {
        fun builder(): ViewTitleBuilder {
            return ViewTitleBuilder()
        }
    }
}
